python/pylibcudf/pylibcudf/io/orc.pyx

```diff
@@ -0,0 +1,283 @@
+            else:
+                self.column_stats["sum"] = None
+        elif ts_stats := get_if[timestamp_statistics](&type_specific_stats):
+            # Before ORC-135, the local timezone offset was included and they were
+            # stored as minimum and maximum. After ORC-135, the timestamp is
+            # adjusted to UTC before being converted to milliseconds and stored
+            # in minimumUtc and maximumUtc.
+            # TODO: Support minimum and maximum by reading writer's local timezone
+            if ts_stats.minimum_utc.has_value() and ts_stats.maximum_utc.has_value():
+                self.column_stats["minimum"] = datetime.datetime.fromtimestamp(
+                    ts_stats.minimum_utc.value() / 1000, datetime.timezone.utc
+                )
+                self.column_stats["maximum"] = datetime.datetime.fromtimestamp(
+                    ts_stats.maximum_utc.value() / 1000, datetime.timezone.utc
+                )
+        else:
+            raise ValueError("Unsupported statistics type")
+
+    def __getitem__(self, item):
+        return self.column_stats[item]
+
+    def __contains__(self, item):
+        return item in self.column_stats
+
+    def get(self, item, alt=None):
+        return self.column_stats.get(item, alt)
+
+    @staticmethod
+    cdef OrcColumnStatistics from_libcudf(column_statistics& col_stats):
+        cdef OrcColumnStatistics out = OrcColumnStatistics.__new__(OrcColumnStatistics)
+        out.number_of_values_c = col_stats.number_of_values
+        out.has_null_c = col_stats.has_null
+        out.type_specific_stats_c = col_stats.type_specific_stats
+        out._init_stats_dict()
+        return out
+
+
+cdef class ParsedOrcStatistics:
+
+    @property
+    def column_names(self):
+        return [name.decode() for name in self.c_obj.column_names]
+
+    @property
+    def file_stats(self):
+        stats_lst = []
+        for i in range(self.c_obj.file_stats.size()):
+            stats_lst.append(OrcColumnStatistics.from_libcudf(self.c_obj.file_stats[i]))
+        return stats_lst
+
+    @property
+    def stripes_stats(self):
+        stats_lst = []
+        for stripe_stats_c in self.c_obj.stripes_stats:
+            stripe_stats = []
+            for i in range(stripe_stats_c.size()):
+                stripe_stats.append(OrcColumnStatistics.from_libcudf(stripe_stats_c[i]))
+            stats_lst.append(stripe_stats)
+        return stats_lst
+
+    @staticmethod
+    cdef ParsedOrcStatistics from_libcudf(parsed_orc_statistics& orc_stats):
+        cdef ParsedOrcStatistics out = ParsedOrcStatistics.__new__(ParsedOrcStatistics)
+        out.c_obj = move(orc_stats)
+        return out
+
+
+cpdef TableWithMetadata read_orc(
+    SourceInfo source_info,
+    list columns = None,
+    list stripes = None,
+    size_type skip_rows = 0,
+    size_type nrows = -1,
+    bool use_index = True,
+    bool use_np_dtypes = True,
+    DataType timestamp_type = DataType(type_id.EMPTY),
```

Comment:
Should this be `None` instead of EMPTY? See how it's used in the implementation, it seems more like an optional.

Reply:
Yes, I think so:

```suggestion
    DataType timestamp_type = None,
```

Reply:
Changed.

Reply:
Not yet pushed?

Reply:
Whoops this just got dropped somehow. Thanks for pointing out.